Sources/Core/Microsoft.StreamProcessing/Operators/Afa/CompiledPartitionedAfaPipe_SingleEvent.cs

```diff
@@ -72,16 +70,16 @@ public override unsafe void OnNext(StreamMessage<TKey, TPayload> batch)
                             var key = srckey[i];
                             var partitionKey = this.getPartitionKey(key);
                             int partitionIndex = EnsurePartition(partitionKey);
-                            var tentativeVisibleTraverser = new FastMap<OutputEvent<TKey, TRegister>>.VisibleTraverser(this.tentativeOutput.entries[partitionIndex].value);
                             long synctime = src_vsync[i];
 
                             if (!this.IsSyncTimeSimultaneityFree)
                             {
                                 int index;
+                                var tentativeVisibleTraverser = new FastMap<OutputEvent<TKey, TRegister>>.VisibleTraverser(this.tentativeOutput.entries[partitionIndex].value);
 
                                 if (synctime > this.lastSyncTime.entries[partitionIndex].value) // move time forward
                                 {
-                                    foreach (var mapIndex in this.seenPartitions.entries[partitionIndex].value) this.seenEvent.Remove(mapIndex);
+                                    this.seenEvent.Remove(srckey[i]);
```

Comment:
Considering srckey[i] is assigned to variable key, it is better to use consistently variable `key` throughout

Reply:
Changed